aiohttp_xmlrpc/handler.py

```diff
@@ -92,7 +92,12 @@ def _check_request(self):
         else:
             kwargs = {}
 
-        result = await asyncio.coroutine(method)(*args, **kwargs)
+        coroutinefunction = (
```

Comment:
It's broken see examples bellow:

```python
# It's not a coroutinefuciton but returns coroutine
def sleeper(sleep_time=0):
    return asuncio.sleep(sleep_time)

# Another example with future
def sleeper(sleep_time=0):
    f = asyncio.Future()
    asyncio.get_event_loop().call_later(sleep_time, f.set_result, True)
    return f
```

Reply:
`asyncio.coroutine` will be removed after 3.10. We should avoid to use it.

Reply:
that's a reason why I wrote the helper in aiormq

Reply:
those arn't broken - they just raise the deprecation warning, to fix them they should be made into async defs:

```python
@staticmethod
async def sleeper(sleep_time=0):
    await asyncio.sleep(sleep_time)

# Another example with future
async def sleeper(sleep_time=0):
    f = asyncio.Future()
    asyncio.get_event_loop().call_later(sleep_time, f.set_result, True)
    return await f
```

Reply:
Of course, adding `async def` helpful, but it's might be incompatible for existent user code. Also `functools.partial` applied to any coroutine function not will be detected by `asyncio.iscoroutinefunction`.

Why you disagree to add simple helper? It brings minimal evil and fast enough on practice.

Reply:
because the aiormq helper is just re-implimenting "asyncio.coroutine". The problem isn't that "asyncio.coroutine" is going away the real problem is using a regular "def" and having it work like an "async def"